libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2608,6 +2610,32 @@ Expression const* SMTEncoder::innermostTuple(Expression const& _expr)
 	return expr;
 }
 
+Expression const* SMTEncoder::cleanExpression(Expression const& _expr)
```

Comment:
Nice! There are probably several places in the code we could cleanup by extending and using this function, but can be done in another  PR.